Panic when creating VMSS

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

```
Terraform v0.12.24
+ provider.azurerm v2.5.0
```


### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_linux_virtual_machine_scale_set`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "azurerm_linux_virtual_machine_scale_set" "at" {
  name                            = "banyan-accesstier-scaleset"
  resource_group_name             = data.azurerm_resource_group.rg.name
  location                        = data.azurerm_resource_group.rg.location
  sku                             = var.sku
  instances                       = var.instance_count
  admin_username                  = var.ssh_user
  disable_password_authentication = true

  admin_ssh_key {
    username   = var.ssh_user
    public_key = var.ssh_public_key
  }

  source_image_reference {
    publisher = "OpenLogic"
    offer     = "CentOS"
    sku       = "7.5"
    version   = "latest"
  }

  os_disk {
    caching              = "ReadWrite"
    storage_account_type = "StandardSSD_LRS"
  }

  network_interface {
    name                      = "banyan-accesstier-nic"
    primary                   = true
    enable_ip_forwarding      = true
    network_security_group_id = azurerm_network_security_group.atsg.id

    ip_configuration {
      name      = "banyan-accesstier-ip"
      primary   = true
      subnet_id = data.azurerm_subnet.subnet.id
      public_ip_address {
        name = "banyan-accesstier-pip"
      }
    }
  }

  boot_diagnostics {
    storage_account_uri = data.azurerm_storage_account.diag.primary_web_endpoint
  }

  custom_data = base64encode(join("", [
    "#!/bin/bash -x\n",
    "systemctl stop firewalld\n",
    "systemctl disable firewalld\n",
    "yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm\n",
    "yum -y install jq tar gzip curl sed yum-utils\n",
    "rpm --import https://www.banyanops.com/onramp/repo/RPM-GPG-KEY-banyan\n",
    "yum-config-manager --add-repo https://www.banyanops.com/onramp/repo\n",
    "yum install -y ${var.package_name}\n",
    "cd /opt/banyan-packages\n",
    "while [ -f /var/run/yum.pid ]; do sleep 1; done\n",
    "BANYAN_ACCESS_TIER=true ",
    "BANYAN_SITE_NAME=${var.site_name} ",
    "BANYAN_SITE_ADDRESS=nlb.bnndemos.com ",
    "BANYAN_SITE_DOMAIN_NAMES=${join(",", var.site_domain_names)} ",
    "BANYAN_SITE_AUTOSCALE=true ",
    "BANYAN_API=${var.api_server} ",
    "BANYAN_HOST_TAGS= ",
    "./install ${var.refresh_token} ${var.cluster_name}\n",
    "semanage port -a -t ssh_port_t -p tcp 2222\n",
    "echo 'Port 2222' >> /etc/ssh/sshd_config && /bin/systemctl restart sshd.service\n"
  ]))
}
```

### Debug Output

https://gist.githubusercontent.com/tradel/4c3284e8ef583f66f2ea31f7267cab72/raw/f86e5ec91d8d7f5c2ca0cb442a4e9749b722b63a/terraform.log

### Panic Output

https://gist.githubusercontent.com/tradel/4c3284e8ef583f66f2ea31f7267cab72/raw/f86e5ec91d8d7f5c2ca0cb442a4e9749b722b63a/crash.log

### Expected Behavior

Create a Linux VMSS.

### Actual Behavior

Provider panicked and Terraform crashed.

```
data.azurerm_resource_group.rg: Refreshing state...
data.azurerm_virtual_network.net: Refreshing state...
data.azurerm_storage_account.diag: Refreshing state...
azurerm_network_security_group.atsg: Refreshing state... [id=/subscriptions/72b1e8b4-f329-494c-bfa6-0a6506006fbb/resourceGroups/CustomerSuccess-USEast/providers/Microsoft.Network/networkSecurityGroups/banyan-accesstier-nsg]
data.azurerm_subnet.subnet: Refreshing state...
azurerm_linux_virtual_machine_scale_set.at: Creating...

Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is nil, not map[string]interface {}
```

### Steps to Reproduce


<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply -auto-approve -var-file=terraform.tfvars`

### Important Factoids

Nothing unusual. My first attempt at Terraforming on Azure, so I may be making an obvious mistake.

--- a/azurerm/internal/services/compute/shared_schema.go
+++ b/azurerm/internal/services/compute/shared_schema.go
@@ -113,7 +113,7 @@ func bootDiagnosticsSchema() *schema.Schema {
 }
 
 func expandBootDiagnostics(input []interface{}) *compute.DiagnosticsProfile {
-	if len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return &compute.DiagnosticsProfile{
 			BootDiagnostics: &compute.BootDiagnostics{
 				Enabled:    utils.Bool(false),

